### DISCUSSION TITLE
Decision Model and Notation (DMN) support

### DISCUSSION THREAD

**Discussion by tokarenko** [imixs/imixs-workflow#790] [General]
Please advise how to invoke Decision Model and Notation (DMN) engine and model the invocation in BPMN for Imixs workflow engine?

With regards, Dmitrii

**Comment by rsoika** [comment]
Hi,
Imixs-Workflow does not support DNN. But there are several ways how to model Decisions in BPMN. 

First of all you can use [conditional Events](https://www.imixs.org/doc/modelling/howto.html#Conditional_Events). In this case you simple place a exclusive Gateway behind a Event and add your conditions to the outgoing sequence flows:
![image](https://user-images.githubusercontent.com/1339124/178021143-f0766f9e-3079-4dc3-8f39-89490884619c.png)

Of course you can also mark one of your flows as a default flow.

The conditions are based on the [Imixs-Workflow Rule Engine.](https://www.imixs.org/doc/core/ruleengine.html). With the latest version the rule engien is based on the script engine of GraalVM which is a extremely fast and flexible script engine. 

The Imixs-Workflow Rule Engine allows you to add any kind of script in a Event to be evaluated during execution. With the [Imixs-Rule-Plugin](https://www.imixs.org/doc/engine/plugins/ruleplugin.html) you can further more use scripts to validate a processing situation or compute new item values.

Let me know if this answers your question.

**Reply by tokarenko** [reply]
Tank you, Ralph, for your comprehensive reply!
